### PR TITLE
Feat/deadletter for subscriptions

### PR DESCRIPTION
**Pull request by jyecusch**
Closes: https://github.com/nitrictech/nitric/issues/287

### CODE REVIEW

File: contracts/proto/faas/v1/faas.proto [hunk 1]
```diff
@@ -71,6 +71,8 @@ message ApiWorker {
 
 message SubscriptionWorker {
   string topic = 1;
+  // optional dead-letter queue where events should be sent if they can't be delivered to/processed by this worker.
+  string dead_letter_queue = 2;
```
Comment (tjholm) [inline contracts/proto/faas/v1/faas.proto:75]: changing `dead_letter_queue` -> `dead_letter` here might be better wording for supporting different sink types for deadletter messages.

File: pkg/api/nitric/v1/document.pb.validate.go [hunk 2]
```diff
@@ -32,7 +31,6 @@ var (
 	_ = (*url.URL)(nil)
 	_ = (*mail.Address)(nil)
 	_ = anypb.Any{}
-	_ = sort.Sort
```
Comment (tjholm) [inline pkg/api/nitric/v1/document.pb.validate.go:35]: @jyecusch can you double check the version of protoc/protoc-gen-go you're using for this? It looks like there is a mismatch with the version we're using for CI/CD. I think @asalkeld implmeneted some install scripts in our makefile for this to ensure the versions are consistent, so it looks like we'll need to double check this.

File: contracts/proto/faas/v1/faas.proto (hunk 1 above)
Comment (jyecusch) [reply contracts/proto/faas/v1/faas.proto]: done

File: pkg/api/nitric/v1/document.pb.validate.go (hunk 2 above)
Comment (jyecusch) [reply pkg/api/nitric/v1/document.pb.validate.go:35]: I have 3.19.4 installed, it looks like the install script is trying to install 3.19.1

**Review by davemooreuws — APPROVED** [review]
LGTM

**Review by asalkeld — COMMENTED** [review]
seems ok to me, but not exactly sure what the intended behavior is

**Pull request merged**